crates/proof-of-sql/src/sql/proof_exprs/range_check.rs

```diff
@@ -0,0 +1,382 @@
+use crate::base::{scalar::Scalar, slice_ops};
+use bumpalo::Bump;
+use bytemuck::cast_slice;
+
+// Decomposes a scalar to requisite words, additionally tracks the total
+// number of occurences of each word for later use in the argument.
+fn decompose_scalar_to_words<'a, S: Scalar + 'a>(
+    scalars: &mut [S],
+    alloc: &'a Bump,
+    word_columns: &mut [&mut [u8]],
+    byte_counts: &mut [u64],
+    byte_rows: &mut Vec<&'a [u8]>,
```

Comment:
We won't need `byte_rows`. You should remove this parameter.

Reply:
I will need byte rows when we need to compute constraints, specifically the sum constraints. I can remove this from this PR, but I will need to add it back later